### PR TITLE
implement the search location screen

### PR DESCRIPTION
**Pull request by daferya**
This pull request introduces several significant changes across multiple files to enhance the functionality and testing of the application. The most notable changes include the addition of a new test class for the `LocationSearchCustomScreen`, updates to the `AnnouncementScreenTest` class, and modifications to the `MainActivity` to integrate the new location search functionality.

### Enhancements to Testing:
* [`app/src/androidTest/java/com/arygm/quickfix/ui/elements/LocationSearchCustomScreenTest.kt`](diffhunk://#diff-e73f42f8b9443ca897113cb8f157ad283e2083a66070190847731df9f1ec6f2fR1-R197): Added a comprehensive test class for `LocationSearchCustomScreen` to verify various functionalities including initial state, search query updates, suggestion clicks, error handling, and back button functionality.
* [`app/src/androidTest/java/com/arygm/quickfix/ui/search/AnnouncementScreenTest.kt`](diffhunk://#diff-640b59c2285c3f61653e59f7cd98593f8969ae4faeb9cb54f8cd54133e593845R4-R20): Enhanced the `AnnouncementScreenTest` class by adding new tests for location box display, updating location selection, handling long text, opening search screen on click, and resetting location after posting an announcement. [[1]](diffhunk://#diff-640b59c2285c3f61653e59f7cd98593f8969ae4faeb9cb54f8cd54133e593845R4-R20) [[2]](diffhunk://#diff-640b59c2285c3f61653e59f7cd98593f8969ae4faeb9cb54f8cd54133e593845L82-L91) [[3]](diffhunk://#diff-640b59c2285c3f61653e59f7cd98593f8969ae4faeb9cb54f8cd54133e593845R113-R130) [[4]](diffhunk://#diff-640b59c2285c3f61653e59f7cd98593f8969ae4faeb9cb54f8cd54133e593845R142-R211)

### Integration of Location Search:
* [`app/src/main/java/com/arygm/quickfix/MainActivity.kt`](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadR35): Integrated `LocationSearchCustomScreen` into the `MainActivity` by adding the necessary imports, updating the `QuickFixApp` function to handle screen changes, and modifying the `SearchNavHost` function to include the new screen. [[1]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadR35) [[2]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadR47) [[3]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadL134-R136) [[4]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadR147) [[5]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadL238-R243) [[6]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadL331-R344) [[7]](diffhunk://#diff-ba2e53eacedc4aa915d40e1d7376dd4303e873d530dc0c808cec05f39c1c9eadR362-R365)

### Dependency Updates:
* [`app/build.gradle.kts`](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R12): Added `kotlin-parcelize` plugin and `gson` library to the project dependencies to support parcelable data classes and JSON parsing. [[1]](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R12) [[2]](diffhunk://#diff-8cff73265af19c059547b76aca8882cbaa3209291406f52df1dafbbc78e80c46R171-R172)

### Data Model Enhancements:
* [`app/src/main/java/com/arygm/quickfix/model/locations/Location.kt`](diffhunk://#diff-e719c0911ac5e574f68889be4a230c4a883095fe3a2ea759f9258683df70c5d8R3-R11): Updated the `Location` data class to implement `Parcelable` using the `kotlin-parcelize` plugin, enabling easier data transfer between components.

### CODE REVIEW

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+                                  locationHelper.getCurrentLocation { currentLocation ->
+                                    Log.e(
+                                        "LocationSearch",
+                                        "kabbour Position actuelle: $currentLocation")
+
+                                    if (currentLocation != null) {
+                                      val userLocation =
+                                          Location(
+                                              name = "Ma position actuelle",
+                                              latitude = currentLocation.latitude,
```
Comment (mohagpt) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt:142]: This should also be in english to be consistent with the rest of the app.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+                                .testTag("use_current_location"),
+                        verticalAlignment = Alignment.CenterVertically) {
+                          MyIcon(screenWidth)
+
+                          Text(
+                              text = "Utiliser ma position actuelle",
+                              style = MaterialTheme.typography.bodyLarge,
+                              color = MaterialTheme.colorScheme.onBackground,
+                              modifier = Modifier.padding(start = 8.dp))
+                        }
```
Comment (mohagpt) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt:164]: This text composable also displays a text in french, it should be in english.

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                  item {
+                    Box(
+                        modifier = Modifier.fillMaxSize().padding(16.dp),
+                        contentAlignment = Alignment.Center) {
+                          Text(
+                              text = "Aucun résultat trouvé",
+                              style = MaterialTheme.typography.bodyLarge.copy(fontSize = 20.sp),
+                              color = MaterialTheme.colorScheme.onBackground,
+                              modifier = Modifier.testTag("no_results_message"))
+                        }
```
Comment (mohagpt) [inline app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt:176]: Same here.

**Review by mohagpt — CHANGES_REQUESTED** [review]
This is a great addition to our project and will definitely enhance its functionality. However, to maintain consistency across the app, it’s important that all displayed text aligns with our standard of using English. Please update the text accordingly so it matches the rest of the app. Overall, excellent work!

File: app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+
+  Scaffold(
+      topBar = {
+        TopAppBar(
+            title = {
+              Box(modifier = Modifier.fillMaxWidth().padding(vertical = 8.dp)) {
+                QuickFixTextFieldCustom(
+                    value = locationQuery,
+                    onValueChange = { locationViewModel.setQuery(it) },
+                    placeHolderText = "Entrer une adresse ou un lieu",
```
[the remark this answers is not in the document]
Comment (daferya) [reply app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixSearchLocation.kt]: Thank you for your feedback. I have made the necessary changes, and the content is now in English. Please let me know if there's anything else you'd like to review. Thanks again!

**Review by mohagpt — APPROVED** [review]
Great work on addressing the requested changes! Everything looks good now, and the implementation aligns well with the project requirements. This new feature is ready to be merged into the main branch. Nice job!

**Pull request merged**